src/sass_types/boolean.cpp

```diff
@@ -15,59 +13,63 @@ namespace SassTypes
     return v ? instance_true : instance_false;
   }
 
-  Handle<Function> Boolean::get_constructor() {
+  v8::Local<v8::Function> Boolean::get_constructor() {
+    Nan::EscapableHandleScope scope;
+    v8::Local<v8::Function> conslocal; 
     if (constructor.IsEmpty()) {
-      Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+      v8::Local<v8::FunctionTemplate> tpl = Nan::New<v8::FunctionTemplate>(New);
 
-      tpl->SetClassName(NanNew("SassBoolean"));
+      tpl->SetClassName(Nan::New("SassBoolean").ToLocalChecked());
       tpl->InstanceTemplate()->SetInternalFieldCount(1);
-      tpl->PrototypeTemplate()->Set(NanNew("getValue"), NanNew<FunctionTemplate>(GetValue)->GetFunction());
+      Nan::SetPrototypeTemplate(tpl, "getValue", Nan::GetFunction(Nan::New<v8::FunctionTemplate>(GetValue)).ToLocalChecked());
 
-      NanAssignPersistent(constructor, tpl->GetFunction());
+      constructor.Reset(Nan::GetFunction(tpl).ToLocalChecked());
+      conslocal = Nan::New(constructor);
 
-      NanAssignPersistent(get_singleton(false).js_object, NanNew(constructor)->NewInstance());
-      NanSetInternalFieldPointer(NanNew(get_singleton(false).js_object), 0, &get_singleton(false));
-      NanNew(constructor)->Set(NanNew("FALSE"), NanNew(get_singleton(false).js_object));
+      get_singleton(false).js_object.Reset(conslocal->NewInstance());
+      Nan::SetInternalFieldPointer(Nan::New(get_singleton(false).js_object), 0, &get_singleton(false));
+      Nan::Set(conslocal, Nan::New("FALSE").ToLocalChecked(), Nan::New(get_singleton(false).js_object));
 
-      NanAssignPersistent(get_singleton(true).js_object, NanNew(constructor)->NewInstance());
-      NanSetInternalFieldPointer(NanNew(get_singleton(true).js_object), 0, &get_singleton(true));
-      NanNew(constructor)->Set(NanNew("TRUE"), NanNew(get_singleton(true).js_object));
+      get_singleton(true).js_object.Reset(conslocal->NewInstance());
+      Nan::SetInternalFieldPointer(Nan::New(get_singleton(true).js_object), 0, &get_singleton(true));
+      Nan::Set(conslocal, Nan::New("TRUE").ToLocalChecked(), Nan::New(get_singleton(true).js_object));
 
       constructor_locked = true;
+    } else {
+      conslocal = Nan::New(constructor);
     }
 
-    return NanNew(constructor);
+    return scope.Escape(conslocal);
   }
 
   Sass_Value* Boolean::get_sass_value() {
     return sass_make_boolean(value);
   }
 
-  Local<Object> Boolean::get_js_object() {
-    return NanNew(this->js_object);
+  v8::Local<v8::Object> Boolean::get_js_object() {
+    Nan::EscapableHandleScope scope;
+    return scope.Escape(Nan::New(this->js_object));
   }
```

Comment:
Since this only creates one new handle, you can safely omit the scope if you inline it.
```c++
inline v8::Local<v8::Object> Boolean::get_js_object() {
  return Nan::New(this->js_object);
}
```